hitch/content.md

```diff
@@ -0,0 +1,63 @@
+# What is Hitch?
+
+[Hitch](https://hitch-tls.org/) is a *libev-based* high performance *SSL/TLS proxy* by [Varnish Software](https://varnish-software.com). It is specifically built to terminate TLS connections at high scale and forwards unencrypted HTTP traffic to Varnish or any other HTTP backend.
+
+# How to use this image
+
+Running a Hitch Docker container can be done by using the following command:
+
+```console
+docker run --name=hitch -p 443:443 hitch:latest
+```
+
+This container will expose port `443`, which is required for HTTPS traffic.
+
+## Configuration file and extra options
+
+Without any argument, the container will run `hitch --config=/etc/hitch/hitch.conf`. You can mount your own configuration file to replace the default one:
+
+```console
+docker run -v /path/to/your/config/file:/etc/hitch/hitch.conf hitch
+```
+
+You can also change the path of the configuration file by setting the `HITCH_CONFIG_FILE` environment variable. You can set it to an empty string to disable the configuration file altogether.
+
+Note that extra arguments can be added to the command line. If the first argument starts with a `-`, the arguments are added to the default command line, otherwise they are treated as a command.
+
+> Our assumption is that your backend, Varnish or other, supports both *HTTP/2* and the *PROXY* protocol.
+
+## Connecting to Varnish
+
+By default Hitch will connect to Varnish using `localhost:8843` using the [PROXY protocol](https://github.com/varnish/hitch/blob/master/docs/proxy-protocol.md). If your `varnishd` process has been started with `-a localhost:8443,PROXY`, the two will be able to talk together and Varnish will expose the true client IP as `client.ip` in VCL.
```

Comment:
It's addressed in the following paragraph, but this feels a little misleading (since running in the same Kubernetes Pod is the only common use case that leads to `localhost` meaning much / working the way this is documented, and even that isn't horribly common these days in my experience since it's so limited :grimacing:).

If this is encoded in the image-provided `hitch.conf` (not the compiled-in defaults), maybe it would make sense to do something like `sed -i -e 's/localhost/varnish/g' /etc/hitch/hitch.conf` so it can work OOTB with multiple containers?

(I think it's also fine to leave as-is -- having users supply a command to set an explicit `--backend` is totally reasonable, but in that case I think this wording could maybe use some tweaking. :smile:)

Reply:
the most common use-case is on the same machine, or in the same pod, so I'd leave `localhost`